crates/core/tedge_watchdog/src/error.rs

```diff
@@ -4,8 +4,11 @@ use tedge_config::{ConfigSettingError, TEdgeConfigError};
 
 #[derive(Debug, thiserror::Error)]
 pub enum WatchdogError {
-    #[error("Fail to run `{cmd}`: {from}")]
-    CommandExecError { cmd: String, from: std::io::Error },
+    #[error(transparent)]
+    SystemdError(systemd::Error),
+
+    #[error("Talking to systemd failed")]
+    TalkingFailed,
```

Comment:
I am not entirely sure what would be appropriate here.
The error should indicate that the underlying systemd library failed to talk to systemd. It is only used with [this call](https://docs.rs/systemd/latest/systemd/daemon/fn.pid_notify.html) right now (documentation refers to [here](https://docs.rs/systemd/latest/systemd/daemon/fn.notify.html)).
And the documentation of latter call says:

> Returns true if systemd was contacted successfully.

So `false` would mean systemd wasn't contacted successfully... and to be honest I don't know what this actually means.

Sure, `SendNotificationFailed` would not be wrong, but it would also not tell what the issue was, so that's why I am struggling with name (and my name, of course).

Reply:
After having another look at [the source of the `notify` fn](https://docs.rs/systemd/latest/src/systemd/daemon.rs.html#309) and [the manpage of `sd_notify`](https://www.freedesktop.org/software/systemd/man/sd_notify.html):


* I believe that the `sd_try!()` macro in the linked source catches the `errno` indicating negative values
* If the message was not send, zero is returned, which would mean we get `Ok(false)`
* Otherwise a positive number, which would mean we get `Ok(true)`

From the manpage of `sd_notify()`:

> If $NOTIFY_SOCKET was not set and hence no status message could be sent, 0 is returned. [...] Note that the return value simply indicates whether the notification message was enqueued properly, it does not reflect whether the message could be processed successfully. Specifically, no error is returned when a file descriptor is attempted to be stored using FDSTORE=1 but the service is not actually configured to permit storing of file descriptors (see above).

So in some way, your suggestion would be okay. :thinking: